racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -0,0 +1,106 @@
+<script setup lang="ts">
+import { useHttp } from '@/plugins/http';
+import { ref, onMounted, inject } from 'vue'
+import SearchBar from '@/components/SearchBar.vue';
+import { injectionKey } from '@/plugins/runtimeConfiguration';
+import type { Ref } from 'vue'
+import type { Datacenter } from './DatacentersView.vue';
+
+const datacenters: Ref<Array<Datacenter>> = ref([])
+const datacenterDetails: Ref<Datacenter | undefined> = ref()
+const racks: Ref<Array<Rack>> = ref([])
+const rackDetails: Ref<Array<Rack>> = ref([])
+const showFullImg = ref(false)
+const http = useHttp()
+
+function openImg(){
+    showFullImg.value = !showFullImg.value
+}
+
+function listInfrastructures(){
+    var rack = rackDetails.value[0].nodes
+    for (let index = 0; index < rack.length; index++) {
+        return rack[index].infrastructure
+    }
+}
+
+async function getRacks(){
+    try {
+        const resp = await http.get('racks')
+        racks.value = resp.data as Rack[]
+        rackDetails.value = racks.value.filter((rack) => rack.room === props.datacenterRoom)
+        console.log(rackDetails.value)
+    } catch (error) {
+        console.error('Erreur lors de la récupératuon des données des datacenters', error)
+    }
+}
+
+async function getDatacenters(){
+    try {
+        const resp = await http.get('datacenters')
+        datacenters.value = resp.data as Datacenter[]
+        datacenterDetails.value = datacenters.value.filter((datacenter) => datacenter.name === props.name)[0]
+    } catch (error) {
+        console.error('Erreur lors de la récupératuon des données des datacenters', error)
+    }
+}
+
+onMounted(() => {
+    getDatacenters()
+    getRacks()
+})
+
+const props = defineProps({
+    name: String,
+    datacenterRoom: String,
+})
+
+interface Rack {
+    room: String,
+    name: String,
+    nodes: [{
+        infrastructure: String,
+    }]
+}
+</script>
+
+<template>
+
+    <SearchBar 
+        v-if="datacenters.length"
+        viewTitle="Datacenter Room"    
+        searchedItem="datacenter" 
+        :items="datacenters"
+    />
+
+    <img :src="`${inject(injectionKey)!.api_server}/draw/room/${ props.datacenterRoom }.svg`" alt="" @click="openImg()" class="h-96 max-w-500 mx-auto p-10">
+    
+    <div v-show="showFullImg" class="fixed top-0 left-0 flex flex-col items-center justify-center w-screen h-screen bg-gray-300 bg-opacity-50 dark:bg-gray-900 dark:bg-opacity-50 backdrop-blur-md z-50">
+        <img :src="`${inject(injectionKey)!.api_server}/draw/room/${ props.datacenterRoom }.svg`" alt="" @click="openImg()" class="h-screen max-w-full bg-white" >
```

Comment:
The URL of the image is used twice, it should be returned by a function to avoid duplication.